src/dash-adapter.js

```diff
@@ -276,18 +306,64 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
             DashAdapter._logger.debug('The source has been loaded successfully');
             resolve(data);
           }).catch((error) => {
-            reject(new Error(
-              error.severity,
-              error.category,
-              error.code,
-              error.data));
+            if (!this._config.forceRedirectExternalStreams &&
+              !this._triedReloadWithRedirect &&
+              this._isManifestLoadError(error)) {
+              this._reloadWithRedirect(startTime, resolve, reject)
+            } else {
+              reject(new Error(
+                error.severity,
+                error.category,
+                error.code,
+                error.data));
+            }
           });
         }
-      });
+      })
+      ;
     }
     return this._loadPromise;
   }
 
+  /**
+   * destroying the adapter and reconfiguring the adapter's config + source member, then calling load again so it can
+   * and reload with the manifest redirect.
+   * @param {number} startTime - Optional time to start the video from.
+   * @param {Function} resolve - original load promise resolve function
+   * @param {Function} reject - original load promise reject function
+   * @returns {void}
+   * @private
+   */
+  _reloadWithRedirect(startTime: ?number, resolve: Function, reject: Function): void {
+    this._triedReloadWithRedirect = true;
+    this._installShakaRedirectScheme();
+    this.reset()
+      .then(this.load)
```

Comment:
`this.load` will bind the correct `this`?
Should we use instead `()=>this.load()` to ensure the correct `this`?